Realm/RLMRealm.mm

```diff
@@ -71,8 +71,7 @@ - (void)notify;
 
 // create NSException from c++ exception
 static __attribute__((noreturn)) void throw_objc_exception(exception &ex) {
```

Comment:
We should just remove this method and throw RLMException wherever this is called.